File: CodeChef/Starters/START200/DIV3/01.cpp
// Problem Link--> https://www.codechef.com/problems/CHEFBAKES77

// Complexity -> Time- O(1) per test case
// Idea //
/*
The maximum number of cakes a truck can carry is floor(y/x)  // Given y>=x so every truck will carry atleast one cake
The number of trucks required will be ceil(n/z) where z is floor(y/x)
In C++ int-division of y/x will directly give floor(y/x) but in python we have to use y//x
To calculate ceil(n/z) = (n+z-1)/z 
*/

// Code -->
#include <bits/stdc++.h>
using namespace std;

int main() {
    // Added just to make the code faster
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
  
    // Input Variables
    int n,x,y;
    cin>>n>>x>>y;
    int z=y/x;

    // Code idea
    cout<<(n+(z-1))/z;
    return 0;
}

File: CodeChef/Starters/START200/DIV3/01.py
"Problem Link -> https://www.codechef.com/problems/CHEFBAKES77"
## Time Complexity -> O(1)

## Idea
"""
The maximum number of cakes a vehicle can carry will be ceil(y/x) 
The minimum number of vehicles required will be floor of (n/z) where z=floor(y/x)
Although python has math library to calculate both floor,ceil.
But it is not advised to use them as for large function they will make code slower
"""

"Formula used"
"Floor (y/x) =(y//x)"
"Ceil (y/x)= (y+x-1)//x"


## Code
"This is a simple code as constraints are very small don't cry on A,B of div3/4 for TLE"
"Constraints are usually that O(n^2) solutions are able to pass and memory limit is not an issue"
## Here using sys.stdin doesn't create a difference because of only one input-output case
n,x,y=map(int,input().split())

z=y//x    ## Calulcating floor(y/x)
print((n+z-1)//z)    ## Calculating ceil(n/z)
